File: src/main/kotlin/ro/bogdannegoita/myplannerkt/domain/TaskProgress.kt
package ro.bogdannegoita.myplannerkt.domain

import ro.bogdannegoita.myplannerkt.commons.TaskDto
import ro.bogdannegoita.myplannerkt.commons.TaskProgressDto
import ro.bogdannegoita.myplannerkt.persistence.daos.TaskProgressDao
import java.util.*

class TaskProgress(data: TaskProgressDto, task: Task, private val dao: TaskProgressDao) {
    val id: UUID = data.id!!
    val completed by data::completed
    val taskData: TaskDto = data.task
}

File: MyPlannerWeb/src/models/TaskProgress.js
import Task from '@/models/Task.js';

export default class TaskProgress {
	id;
	completed;
	task;

	constructor(taskProgress) {
		if (!taskProgress) return;
		this.id = taskProgress.id;
		this.completed = taskProgress.completed;
		this.task = new Task(taskProgress.task);
	}
}
